ethcore/sync/src/chain/supplier.rs

```diff
@@ -215,6 +221,10 @@ impl SyncSupplier {
 			if let Some(body) = io.chain().block_body(BlockId::Hash(r.val_at::<H256>(i)?)) {
 				data.append(&mut body.into_inner());
 				added += 1;
+				// Check that the packet won't be oversized
+				if data.len() > payload_soft_limit {
```

Comment:
Same as with headers, not sure how reliable is `RlpStream::len`

Reply:
Here `data` is of type `Bytes`